src/NUnitFramework/tests/Internal/Filters/NotFilterTests.cs

```diff
@@ -67,7 +67,7 @@ public void ExplicitMatchTest()
         {
             var filter = new NotFilter(new CategoryFilter("Dummy"));
 
-            Assert.False(filter.IsExplicitMatch(_topLevelSuite));
+            Assert.True(filter.IsExplicitMatch(_topLevelSuite));
```

Comment:
We may need to come back to this in a separate issue since this wasn't really the point of the bug you're fixing.

Explicit was invented to avoid anything being run unless it was specifically selected by the _presence_ of a particular name or category, not by it's _absence_. So, there's some logic to saying that a not filter should always return false. We would need a bunch of tests to define this clearly though.